app/root.tsx

```diff
@@ -1,8 +1,68 @@
-import { Links, Meta, Outlet, Scripts } from "react-router";
+import { useAtom } from "jotai";
+import { atomWithStorage } from "jotai/utils";
+import {
+  Links,
+  LoaderFunctionArgs,
+  Meta,
+  Outlet,
+  Scripts,
+  useLoaderData,
+} from "react-router";
 import { SideBar } from "~/components/sideBar";
+import { account } from "~/lib/account";
+import { accountCookie } from "~/lib/login";
+import { parseToken } from "~/lib/parseToken";
 import styles from "~/root.module.css";
 
-export function Layout({ children }: { children: React.ReactNode }) {
+export interface LoggedInAccountDatum {
+  id: string;
+  name: string;
+  nickname: string;
+  avatarURL: string;
+  headerURL: string;
+}
+export const loggedInAccountAtom = atomWithStorage<
+  LoggedInAccountDatum | undefined
+>("loggedInAccount", undefined);
```

Comment:
```suggestion
export const loggedInAccountAtom = atomWithStorage<
  LoggedInAccountDatum | undefined
>("loggedInAccount", undefined, undefined, { getOnInit: true });
```
[`getOnInit` オプション](https://jotai.org/docs/utilities/storage#parameters) を有効にすれば, 初期化時に `undefined` ではなく `localStorage` 側のデータを取ってこれるようです.